boa/src/exec/mod.rs

```diff
@@ -424,8 +424,10 @@ impl Interpreter {
     /// https://tc39.es/ecma262/#sec-toobject
     #[allow(clippy::wrong_self_convention)]
     pub(crate) fn to_object(&mut self, value: &Value) -> ResultValue {
-        match value {
-            Value::Undefined | Value::Null => Err(Value::undefined()),
+    match value.data() {
```

Comment:
```suggestion
    match value {
```

#498 merged `Value` and `ValueData` together so we dont have to call `.data()`, this should fix it :)